File: cmd/66_plus_one.go
package cmd

func plusOne(digits []int) []int {
    l := len(digits)
    for i, _ := range (digits) {
        last := digits[l-1-i]
        if last + 1 < 10 {
            digits[l-1-i] = last + 1
            break
        } else {
            digits[l-1-i] = 0
            if (l-1-i == 0) {
                digits=append([]int{1},digits...)
            }
        }
    }
    return digits
    
}

File: cmd/66-PlusOne/66-PlusOne.cs
public class Solution {
    public int[] PlusOne(int[] digits)  {
        for (int i= digits.Length -1; i>=0; i--  ) {
            if( digits[i]+1 < 10) {
                digits[i] += 1;
            break;
            }
            else {
                digits[i]=0;
                if (i==0){
                    int[] newDigits = new int[digits.Length + 1];
newDigits[0] = 1;                                // set the prepended value
Array.Copy(digits, 0, newDigits, 1, digits.Length); // copy the old values
return newDigits;
                }    
            }
        }
        return digits;
    }
}